src/RA_Defs.h

```diff
@@ -225,30 +255,382 @@ typedef DWORD			ARGB;
 #undef ASSERT
 #define ASSERT( x ) {}
 #endif
-	
+
 #ifndef UNUSED
 #define UNUSED( x ) ( x );
 #endif
 
-extern std::string Narrow(const wchar_t* wstr);
-extern std::string Narrow(const std::wstring& wstr);
-extern std::wstring Widen(const char* str);
-extern std::wstring Widen(const std::string& str);
 
-//	No-ops to help convert:
-//	No-ops to help convert:
-extern std::wstring Widen(const wchar_t* wstr);
-extern std::wstring Widen(const std::wstring& wstr);
-extern std::string Narrow(const char* str);
-extern std::string Narrow(const std::string& wstr);
 
-typedef std::basic_string<TCHAR> tstring;
 
 
+
+
+
+namespace ra {
+using cstring       = const char*;
+using cwstring      = const wchar_t*;
+using tstring       = std::basic_string<TCHAR>;
+
+template<typename CharT>
+using string_t = std::basic_string<CharT>;
+
+using DataStream     = std::basic_string<BYTE>;
+using ustring        = DataStream;
+using uofstream      = std::basic_ofstream<BYTE>;
+using uifstream      = std::basic_ifstream<BYTE>;
+using uostringstream = std::basic_ostringstream<BYTE>;
+using uistringstream = std::basic_istringstream<BYTE>;
+using uostream       = std::basic_ostream<BYTE>;
+using uistream       = std::basic_istream<BYTE>;
```

Comment:
What are all of these `u` data types? Please don't define things until we need to use them.

Reply:
These are so we don't need to convert back and forth between std::string and DataStream. We're not using std::u16string/u32string but it's still there.

Reply:
if these are just for the conversion, they should go in the .cpp file where the conversion function exists.

Reply:
Sure, I see what what you mean.